sdk/search/azure-search-documents/azure/search/documents/indexes/models/_models.py

```diff
@@ -590,3 +944,13 @@ def unpack_analyzer(analyzer):
             analyzer
         )
     return analyzer
+
+
+def validate(kwargs, version, unsupported):
+    unsupported = [unsupported] if isinstance(unsupported, str) else unsupported
+    errors = []
+    for param in unsupported:
+        if param in kwargs:
+            errors.append(param)
+    if errors:
+        raise ValueError("Unsupported parameters for skill version {}: {}".format(version, ', '.join(errors)))
```

Comment:
It looks like a breaking change.

In old behavior, non-supported parameters are ignored rather than raise exception in ctor.

Reply:
My understanding was you wanted an error thrown:

> This is from Alan said: if user calls setter for non-V1 properties in V1 model, they raise exceptions.
> 
> And it looks to me this is like the scenario user wants to access non-existing properties.
> 
> We can double check with Johan/Anna.

Reply:
It is in property's setter (and maybe getter as well). But if we add it in ctor, it is a breaking change.

Reply:
@annatisch can you provide guidance here? It seems odd to me to just silently accept bad params in the constructor but fail on a setter/getter. It seems we should either always validate or never validate.

Reply:
I am afraid the answer is NO. 

Service team may add/remove properties so it is possible V3 has unsupported properties against V4 and/or V4 has unsupported properties against V3.

Reply:
In that case, a customer is always going to be broken if they're using LATEST. It's just a matter of where the error is raised - either at the model or at the operation.
We also need to accommodate the fact that people may not use models, and pass in their inputs as dicts, in which case, again, the error will be at the operation level.

Reply:
I'll work on moving the validation into the operation.

Reply:
Per our Teams discussion, I've removed validation entirely for these release. We can align on the desired behavior, because the service does not throw when unsupported kwargs are passed.